python/pyspark/sql/types.py

```diff
@@ -1119,8 +1120,8 @@ def Dict(d):
     class Row(tuple):
 
         """ Row in DataFrame """
-        __DATATYPE__ = dataType
-        __FIELDS__ = tuple(f.name for f in dataType.fields)
+        __datatype = dataType
```

Comment:
Ah, gotcha.  Do you want to go ahead with the renaming here to eliminate the double-reference?